File: ch13/Chapter13-WCFService-VB/App_Code/Calculator.vb
' NOTE: You can use the "Rename" command on the context menu to change the class name "Calculator" in code, svc and config file together.
Public Class Calculator
    Implements ICalculator

    Public Function Add(a As Integer, b As Integer) As Integer Implements ICalculator.Add
        Return (a + b)
    End Function

    Public Function Divide(a As Integer, b As Integer) As Integer Implements ICalculator.Divide
        Return (a / b)
    End Function

    Public Function Multiply(a As Integer, b As Integer) As Integer Implements ICalculator.Multiply
        Return (a * b)
    End Function

    Public Function Subtract(a As Integer, b As Integer) As Integer Implements ICalculator.Subtract
        Return (a - b)
    End Function
End Class

File: ch13/Chapter13-WCFService-CS/App_Code/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.Text;

// NOTE: You can use the "Rename" command on the "Refactor" menu to change the class name "Calculator" in code, svc and config file together.
public class Calculator : ICalculator
{
    public int Add(int a, int b)
    {
        return (a + b);
    }

    public int Subtract(int a, int b)
    {
        return (a - b);
    }

    public int Multiply(int a, int b)
    {
        return (a * b);
    }

    public int Divide(int a, int b)
    {
        return (a / b);
    }
}